### ISSUE TITLE
Problem with reading values from TJuiDialog

### ISSUE THREAD

**Issue by szroman** [pradosoft/prado#588]
Hi,
It is possible to read ClientId from TextBox, but I can not read Text or SafeText value.

Example:
```php
class Home extends TPage
{
        public function ButtonClicked($sender, $param)
        {
                if ($this->Page->IsValid)
                {
                    $this->client_id->Text = $this->field->ClientId;
                    $this->client_value->Text = $this->field->SafeText;

                    $this->dialog->close();
                }
        }
}
```
```html
<com:TContent ID="Main">

    <com:TJuiDialog
        ID="dialog"
        Options.Title="Test dialog"
        Options.AutoOpen="true"
        Options.Width="800"
        Options.Modal="true">

        <com:TActiveTextBox ID="field" />

        <com:TActiveButton
            Text="click"
            OnClick="ButtonClicked">
        </com:TActiveButton>

    </com:TJuiDialog>

    <com:TActiveLabel ID="client_id" /> :
    <com:TActiveLabel ID="client_value" />

</com:TContent>
```

**Comment by ctrlaltca** [comment]
This is a limitation inherited from jQuery-ui, have a look at eg. http://stackoverflow.com/questions/1955879/jquery-dialog-posting-of-form-fields
The dialog removes its inner elements from the <form> and adds them back on the root element of the body to ensure it has no positioning problems, so when we serialize() the form at https://github.com/pradosoft/prado/blob/master/framework/Web/Javascripts/source/prado/activecontrols/ajax3.js#L301 , the <input> is not found.

There are a few workarounds that we can try:
```js
// re-append the dialog inside the form after creating it -> potential positioning problems
jQuery("#test").dialog({...}).parent().appendTo("form");
```
or 
```js
// serialize all the input, select, textarea tags even if they are outside the form
jQuery('input, select, textarea').serialize()
```

**Comment by szroman** [comment]
Thanks for your quick response. I am not a javascript programmer and it is not clear for me. Could you show me how to use it in my example ? I am using NModalPanel in my application and I wanted to move to TJuiDialog. I am using complicated structures in my NModalPanel windows with TActiveRepeater mixed with TActiveDataList and others. Will it all work in TJuiDialog after applying your workarounds ?

**Comment by szroman** [comment]
**Help ! Is anybody out there ?**

**Comment by ctrlaltca** [comment]
Both methods require a modification to Prado's javascript files, and thus some testing to ensure nothing else gets broken by this change.

**Comment by ctrlaltca** [comment]
If you want to be a guinea pig for this change, just open ```framework/Web/Javascripts/source/prado/activecontrols/ajax3.js``` and replace line 301:
```
// old
return jQuery(form).serialize() + '&' + jQuery.param(data);
// new
return jQuery('input, select, textarea').serialize() + '&' + jQuery.param(data);
```

**Comment by szroman** [comment]
Thanks a lot for your help. It seems to work flawlessly. My employees are testing it now. If we won't find any problem I will close this issue. Thanks again.

**Comment by ctrlaltca** [comment]
I'd keep this open as a remainder to test the proposed solution on a few corner cases and eventually include it in prado.

**Comment by ctrlaltca** [comment]
Fixed by #593
